### PR TITLE
Migrate icons to design system AB#10356

### PR DESCRIPTION
**Pull request by MikeLyttle**
# Implements [AB#10356](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/10356)

-   [x] Enhancement
-   [ ] Bug
-   [ ] Documentation

## Description

- Adds a new hg-icon component and 3 approved sizes for icons: small, medium and large
- Updates all icons on the site to utilize the new component and match one of the approved sizes
- Applies a standard margin to icons that are placed inline with text

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

-   [x] Unit Tests Updated
-   [ ] Functional Tests Updated
-   [ ] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

YES

### Browsers Tested

-   [x] Chrome
-   [ ] Safari
-   [ ] Edge
-   [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant. Include any specialized deployment steps here.

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/src/components/shared/pageTitle.vue [hunk 1]
```diff
@@ -3,7 +3,7 @@ import Vue from "vue";
 import { Component, Prop } from "vue-property-decorator";
 
 @Component
-export default class PageTitleComponent extends Vue {
+export default class PageTitle extends Vue {
```
Comment (tiagograf-AOT) [inline Apps/WebClient/src/ClientApp/src/components/shared/pageTitle.vue:6]: why are we dropping the "component" word?

File: Apps/WebClient/src/ClientApp/src/components/shared/statusLabel.vue [hunk 2]
```diff
@@ -3,7 +3,7 @@ import Vue from "vue";
 import { Component, Prop } from "vue-property-decorator";
 
 @Component
-export default class StatusLabelComponent extends Vue {
+export default class StatusLabel extends Vue {
```
Comment (tiagograf-AOT) [inline Apps/WebClient/src/ClientApp/src/components/shared/statusLabel.vue:6]: same /\

File: Apps/WebClient/src/ClientApp/src/components/shared/pageTitle.vue (hunk 1 above)
Comment (MikeLyttle) [reply Apps/WebClient/src/ClientApp/src/components/shared/pageTitle.vue:6]: I discussed this with @FuriousLlama and he thought that \<BaseButtonComponent\> was too wordy and we should also prefer kebab case when using it, so I standardized this across the other shared components I'd created as well.

Comment (FuriousLlama) [reply Apps/WebClient/src/ClientApp/src/components/shared/pageTitle.vue:6]: I think there was a miss understanding. I believe we should keep the 'Component' part on the name of the class but not on the aliasing of the components when used

Comment (MikeLyttle) [reply Apps/WebClient/src/ClientApp/src/components/shared/pageTitle.vue:6]: Understood.  I can revert that change.

**Review by tiagograf-AOT — DISMISSED** [review]

File: Apps/WebClient/src/ClientApp/src/components/modal/noteEdit.vue [hunk 3]
```diff
@@ -247,10 +238,7 @@ export default class NoteEditComponent extends Vue {
             <b-row class="w-100 h-100">
                 <b-col cols="auto">
                     <div class="icon">
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/src/components/modal/noteEdit.vue:240]: nit: these should be renamed to icon-wrapper or something along those lines since they are not actually classes for icons, but for the wrappers for them

**Review by FuriousLlama — DISMISSED** [review]

File: Apps/WebClient/src/ClientApp/src/components/modal/noteEdit.vue (hunk 3 above)
Comment (MikeLyttle) [reply Apps/WebClient/src/ClientApp/src/components/modal/noteEdit.vue]: Fixed in latest commit.

**Review by tiagograf-AOT — APPROVED** [review]

**Review by FuriousLlama — APPROVED** [review]
